### ISSUE TITLE
*Accept* does not return the original follow, but a new one with a wrong id

### ISSUE THREAD

**Issue by qwazix** [pixelfed/pixelfed#1710]
I'm requesting a follow from [pherephone](https://github.com/writeas/pherephone) with the following json
``` json
{
        "@context": [
                "https://www.w3.org/ns/activitystreams"
        ],
        "actor": "https://floorb.qwazix.com/activityserve_test_actor_3",
        "id": "https://floorb.qwazix.com/activityserve_test_actor_3/item/zEOMuDhUDV5wnxFC",
        "object": "https://pixelfed.social/users/qwazix",
        "type": "Follow"
} 
```

When pixelfed returns the `Accept` activity, the `object` is not the original follow but this:
``` json
{
    "actor":"https://floorb.qwazix.com/activityserve_test_actor_3",
    "id":"https://floorb.qwazix.com/activityserve_test_actor_3",
    "object":"https://pixelfed.social/users/qwazix",
    "type":"Follow"
}
```

I can work around the issue but I'd really like to be able to compare id's so that I know I'm the one that requested the follow and a malicious actor that sprays accepts doesn't end up getting followed.

**Comment by dansup** [comment]
@qwazix As far as I know, this is how Mastodon handles this. It works with Mastodon and Pleroma and we have no intention of changing this.

**Comment by trwnh** [comment]
@dansup mastodon uses anchors for follow/like activities. it is incorrect behavior to copy the root document id alone without the other fragments, and honestly it shouldn't be that hard to just copy the correct id out of the original Follow activity...

**Comment by qwazix** [comment]
Thanks :)

**Comment by dansup** [comment]
Fixed in #1717 !
